pyvista/core/dataset.py

```diff
@@ -2552,3 +2552,33 @@ def cell_type(self, ind: int) -> int:
 
         """
         return self.GetCellType(ind)
+
+    def cell_point_ids(self, ind: int) -> List[int]:
+        """Return the point ids in a cell.
+
+        Parameters
+        ----------
+        ind : int
+            Cell ID.
+
+        Returns
+        -------
+        list[int]
```

Comment:
Actually, one question too: should we extend this to sequence inputs, returning a list of lists? Some cell/point-related filters do that, then again those are probably different (return meshes, probably). I don't know if this would make sense for use cases.

Reply:
@adeak, are you suggesting a use case like:

(not actually implemented)
```py
>>> mesh.cell_point_ids([0, 3, 5])
[[0, 1, 2], [0, 4, 6], [8, 10, 12]]
```

Reply:
I'd like to include that here if there's a clean way to vectorize this.

Further, I think it'd be nice to expand many of these methods that accept point/cell IDs to accept slices and even `Elipses` (`...`) -- but this is worthy of a whole separate PR and discussion

Reply:
I think we should think through what is the ideal API design before implementing here.  One thought we discussed is to have an API that exposes a wrapper for `vtkCell`.  We could have a property that allows access to the cells themselves, for example `mesh.cells[0]`.  Then, we could do something like `[cell.point_ids for cell in mesh.cells[0:5]]`.  Another thought is, like you suggest, to allow more flexibility in the existing methods.

Reply:
> @adeak, are you suggesting a use case like

Yup.

> I'd like to include that here if there's a clean way to vectorize this.

I doubt it, because cells can have different sizes, so thebresult can't be a (decent) 2d array in general.

Reply:
To close this up, I think this is fine for now. We will either supersede all this with a full-fledged cells attribute (or similar) making this method probably obsolete, or we will keep this around and we can extend it to the sequence-input case later. In other words, I don't think we'd need this method name for something fundamentally different in the future, so we can add this as-is to cover the linked issue and then see where things go.

Reply:
I agree, for future consideration I plan on submitting a method for checking whether a point is inside a cell.  You could imagine providing either a sequence of cells or a sequence of points.  Does it makes sense to allow both as sequences?

Reply:
Hmm, good question. I can't tell how common that use case might be, but when someone _does_ need it, having an implementation for that would be useful. Getting the logistics right for a relatively efficient implementation might make that worthwhile.